checkpoint.go

```diff
@@ -106,6 +107,12 @@ type Checker struct {
 	nextCheckAtLock sync.RWMutex
 }
 
+// Flag is an attribute of a product.
```

Comment:
Nitpick: I guess this type is named `Flag` because we had `Flags map[string]string` in first place, but this comment suggests it should maybe be called otherwise? e.g. `ProductAttribute`? `Attribute`?